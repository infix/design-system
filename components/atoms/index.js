import Button from './Button'
import Input from './Input'
import Heading from './Heading'
import Text from './Text'
import IconButton from './IconButton'
import { RadioGroup, RadioButton } from './Radio'
import { Select } from './Select'

import { TwitterShareButton, FacebookShareButton } from './ShareButton'

const atoms = {
  Text,
  Heading,
  Button,
  IconButton,
  Input,
  RadioGroup,
  RadioButton,
  Select,
  TwitterShareButton,
  FacebookShareButton
}

export default atoms
