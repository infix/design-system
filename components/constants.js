/* global module */
const defaultWeights = [
  400,
  700
]

const defaultBreakpoints = [
  40,
  52,
  64
]

const defaultSpace = [
  0,
  8,
  16,
  32,
  64
]

const defaultFontSizes = [
  12,
  14,
  16,
  20,
  24,
  32,
  48,
  64,
  72
]

module.exports = {
  defaultBreakpoints,
  defaultSpace,
  defaultFontSizes,
  defaultWeights
}
