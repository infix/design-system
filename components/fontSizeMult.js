/* global require */
import {
  style,
  px
} from 'styled-system'

const {
  defaultFontSizes
} = require('./constants')

export const fontSizeMult = (propName, k) => {
  /* eslint-disable no-console */
  console.error('usage of fontSizeMult will be deprecated')
  /* eslint-enable no-console */

  return style({
    prop: 'fontSize',
    cssProperty: propName,
    key: 'fontSizes',
    transformValue: n => px(n * k),
    scale: defaultFontSizes
  })
}

export default fontSizeMult
