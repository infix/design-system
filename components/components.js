import atoms from './atoms'
import molecules from './molecules'
import organisms from './organisms'

import BarChart from './charts/BarChart'
import PieChart from './charts/PieChart'

/* OONI produced atoms */

export const Button = atoms.Button
export const Input = atoms.Input
export const RadioButton = atoms.RadioButton
export const RadioGroup = atoms.RadioGroup
export const TwitterShareButton = atoms.TwitterShareButton
export const FacebookShareButton = atoms.FacebookShareButton
export const Heading = atoms.Heading
export const Text = atoms.Text
export const Select = atoms.Select

/* OONI produced molecules */
export const InputWithIconButton = molecules.InputWithIconButton
export const LogoOONIRun = molecules.LogoOONIRun
export const Card = molecules.Card

/* OONI produced organisms */
export const Hero = organisms.Hero
export const HeroLead = organisms.HeroLead
export const Modal = organisms.Modal

/* Here we just export rebass until we implement all of these */
import {
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Checkbox,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Label,
  Lead,
  Link,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Radio,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Slider,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Textarea,
  Toolbar,
  Tooltip,
  Truncate
} from 'rebass'

export { default as BarChart } from './charts/BarChart'
export { default as PieChart } from './charts/PieChart'

const components = {
  Button,
  Heading,
  Text,
  Input,
  Select,
  Card,
  RadioButton,
  RadioGroup,
  InputWithIconButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart,
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Checkbox,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Label,
  Lead,
  Link,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Radio,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Slider,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Textarea,
  Toolbar,
  Tooltip,
  Truncate
}

export default components
