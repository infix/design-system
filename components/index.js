import {
  Provider,
  Flex,
  Box
} from 'rebass'

import colors from './theme/colors'
import theme from './theme/rebassTheme'

export { default as colors } from './theme/colors'
export { default as theme } from './theme/rebassTheme'

import components from './components'

// We export these from rebass directly
export {
  Provider,
  Flex,
  Box
} from 'rebass'

export const {
  Button,
  ButtonOutline,
  ButtonCircle,
  ButtonTransparent,
  Link,
  NavLink,
  BlockLink,
  Heading,
  Subhead,
  Text,
  Small,
  Lead,
  Pre,
  Code,
  Samp,
  Blockquote,
  Measure,
  Truncate,
  Label,
  Input,
  Select,
  Textarea,
  Checkbox,
  Radio,
  Slider,
  Image,
  Avatar,
  BackgroundImage,
  Container,
  Divider,
  Border,
  Media,
  Card,
  Banner,
  Panel,
  PanelHeader,
  PanelFooter,
  Progress,
  Message,
  Group,
  Toolbar,
  Badge,
  Circle,
  Tabs,
  TabItem,
  DotButton,
  Close,
  Relative,
  Absolute,
  Fixed,
  Sticky,
  Drawer,
  Overlay,
  Carousel,
  ScrollCarousel,
  CarouselSlide,
  Tooltip,
  Switch,
  Arrow,
  Star,
  Embed,
  Donut,
  Row,
  Column,
  RadioGroup,
  RadioButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  InputWithIconButton,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart
} = components

export const OONIComponents = Object.assign({}, components, { Provider, Box, Flex, colors, theme })

export default OONIComponents
