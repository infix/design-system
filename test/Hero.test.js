/* global describe, test, expect */
import React from 'react'
import { renderJSON } from './index'
import { Hero } from '../components'

describe('Hero', () => {
  test('renders', () => {
    const json = renderJSON(
      <Hero />
    )
    expect(json).toMatchSnapshot()
  })

  test('renders with children', () => {
    const json = renderJSON(
      <Hero>
        <div />
      </Hero>
    )
    expect(json).toMatchSnapshot()
  })
})
