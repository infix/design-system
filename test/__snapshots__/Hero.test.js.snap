// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders 1`] = `
.c2 {
  box-sizing: border-box;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
}

.c1 {
  max-width: 1024px;
}

.c0 {
  font-size: 28px;
  width: 100%;
  background-color: #0588cb;
  color: #f8f9fa;
  text-align: center;
}

<Hero
  fontSize={3}
>
  <Hero__StyledHero
    fontSize={3}
  >
    <div
      className="c0"
      fontSize={3}
    >
      <ForwardRef
        extend={[Function]}
        maxWidth={1024}
        mx="auto"
        px={3}
      >
        <ForwardRef
          className="c1"
          extend={[Function]}
          maxWidth={1024}
          mx="auto"
          px={3}
        >
          <Box
            className="c1"
            maxWidth={1024}
            mx="auto"
            px={3}
          >
            <div
              className="c1 c2"
            />
          </Box>
        </ForwardRef>
      </ForwardRef>
    </div>
  </Hero__StyledHero>
</Hero>
`;

exports[`Hero renders with children 1`] = `
.c2 {
  box-sizing: border-box;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
}

.c1 {
  max-width: 1024px;
}

.c0 {
  font-size: 28px;
  width: 100%;
  background-color: #0588cb;
  color: #f8f9fa;
  text-align: center;
}

<Hero
  fontSize={3}
>
  <Hero__StyledHero
    fontSize={3}
  >
    <div
      className="c0"
      fontSize={3}
    >
      <ForwardRef
        extend={[Function]}
        maxWidth={1024}
        mx="auto"
        px={3}
      >
        <ForwardRef
          className="c1"
          extend={[Function]}
          maxWidth={1024}
          mx="auto"
          px={3}
        >
          <Box
            className="c1"
            maxWidth={1024}
            mx="auto"
            px={3}
          >
            <div
              className="c1 c2"
            >
              <div
                key=".0"
              />
            </div>
          </Box>
        </ForwardRef>
      </ForwardRef>
    </div>
  </Hero__StyledHero>
</Hero>
`;
